lib/elixir/lib/inspect/algebra.ex

```diff
@@ -189,6 +189,7 @@ defmodule Inspect.Algebra do
   @container_separator ","
   @tail_separator " |"
   @newline "\n"
+  @interpolation_identifier "\#{"
```

Comment:
Inspect.Algebra is general purpose and it should not know about the formatter. :) We also need to make sure that any expression inside the interpolation doesn't change either.

I think you can solve this fully inside the formatter by calling `doc |> format_to_string |> string`  in the formatter where `doc` is the interpolation contents.

Reply:
@josevalim yeah makes sense to keep it in the formatter, not sure why but when I was initially following the code I couldn't find `\#{` in the formatter 😆 

Gave this another shot, I initially tried doing the below but it didn't seem to solve the issue (from your explanation it seemed you recommended trying to do it on the contents of the interpolation which is why I tried this)
```
doc = doc |> format_to_string() |> string()
doc = surround("\#{", doc, "}")
``` 

Next I tried doing the below, but this had a similar issue to my first attempt where it messed up the comments in the interpolation and put everything on one line
```
surround("\#{", doc, "}") |> format_to_string() |> string()
``` 

So I took a look at what `doc` was and noticed a `{:doc_force, ...}` at the start of the algebra in the comment inside the interpolation case so I had to handle the edge case of not putting it all on one line. Let me know if you have any better ideas!